### DISCUSSION TITLE
Crawl data

### DISCUSSION THREAD

**Discussion by TramBuiH** [NhanDoV/Kaggle-6-first-projects#3] [Q&A] [answered]
Excuse me, I was new bee with Python, my friend showed me your Github to me, where I have read this [your notebook](https://github.com/Nhan121/Kaggle-6-first-projects/blob/master/web-scraping/selenium%2Bbs4/crawl-myfpt.ipynb), and I want to grasp something below:

1. How did you know the class `"h3YV2d"` is used in the comment / reviews from this page-source
`content = BeautifulSoup(res.content, "html.parser")`
`cmt_divs = content.find_all("div", {"class": "h3YV2d"})`

3. Next, I dont understand this
`button = '//*[@id="yDmH0d"]/c-wiz[2]/div/div/div[1]/div[2]/div/div[1]/c-wiz[4]/section/div/div/div[5]/div/div/button/span'`
can you explain me how to get this value?

Thanks

**Comment by NhanDoV** [comment, marked as answer]
Thank for your question, this one I don't want to use English in [this notebook](https://github.com/Nhan121/Kaggle-6-first-projects/blob/master/web-scraping/selenium%2Bbs4/crawl-myfpt.ipynb). To answer this, 

1 .We have 2 ways to know this, 
- Print out the `content` then find at the text (you want) from the page-source
- Press F12, find the section from the page-source which contain the element you want to find (review, datetime, user-name, etc.)
2. For understand how to extract an element from a web using Selenium, you should view its source-page (again press F12) then select the element from the web. Right-click on the element and select `Copy XPath` or anything you want (e.g. CSS_SELECTOR, etc.)

You can refer another notebook [here](https://github.com/Nhan121/Kaggle-6-first-projects/blob/master/web-scraping/chrome_selenium_linkedin/web-scrap-selenium-linkedin-url.ipynb). 

Again, thank you for your considerations.

**Reply by TramBuiH** [reply]
> Thank for your question, this one I don't want to use English in [this notebook](https://github.com/Nhan121/Kaggle-6-first-projects/blob/master/web-scraping/selenium%2Bbs4/crawl-myfpt.ipynb). To answer this,
> 
> 1 .We have 2 ways to know this,
> 
> * Print out the `content` then find at the text (you want) from the page-source
> * Press F12, find the section from the page-source which contain the element you want to find (review, datetime, user-name, etc.)
> 
> 2. For understand how to extract an element from a web using Selenium, you should view its source-page (again press F12) then select the element from the web. Right-click on the element and select `Copy XPath` or anything you want (e.g. CSS_SELECTOR, etc.)
> 
> You can refer another notebook [here](https://github.com/Nhan121/Kaggle-6-first-projects/blob/master/web-scraping/chrome_selenium_linkedin/web-scrap-selenium-linkedin-url.ipynb).
> 
> Again, thank you for your considerations.
--------------------------------------
Thank Nhan, but I have mimic your code (from [line 15](https://github.com/Nhan121/Kaggle-6-first-projects/blob/master/web-scraping/selenium%2Bbs4/crawl-myfpt.ipynb) and I can not crawl all the data like you
![image](https://user-images.githubusercontent.com/115329619/194687151-1cbc92c3-118a-4c60-9907-97a17f947e38.png)

for me, just 43 records
![image](https://user-images.githubusercontent.com/115329619/194687280-fba5c8be-6680-4cd4-a21e-d39d0fa062aa.png)

**Reply by TramBuiH** [reply]
Thanks ^^

**Reply by NhanDoV** [reply]
yeah! I think you need to wait a lot when using the scroll-down button from selenium before the page load more content :D

**Comment by VanVan1234** [comment]
great :D
